core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/WorkflowSubmissionActor.scala

```diff
@@ -237,8 +243,11 @@ trait WorkflowSubmission extends FutureSupport with LazyLogging with MethodWiths
   def submitWorkflowBatch(batch: WorkflowBatch)(implicit executionContext: ExecutionContext): Future[WorkflowSubmissionMessage] = {
 
     val WorkflowBatch(workflowIds, submissionRec, workspaceRec) = batch
+
     // implicitly passed to WorkflowComponent methods which update status
-    implicit val wfStatusCounter = workflowStatusCounter(workspaceSubmissionMetricBuilder(workspaceRec.toWorkspaceName, submissionRec.id))
+    implicit val wfStatusCounter = (status: WorkflowStatus) =>
+      if (trackDetailedSubmissionMetrics) Some(workflowStatusCounter(workspaceSubmissionMetricBuilder(workspaceRec.toWorkspaceName, submissionRec.id))(status))
```

Comment:
nit pick - use Option instead of Some directly to protect against possible Some(null)